lib/dirac_domain_wall_4d.cpp

```diff
@@ -68,10 +69,10 @@ namespace quda {
     blas::xpay(in, -kappa5, out);
   }
 
-  void DiracDomainWall4D::MdagM(ColorSpinorField &out, const ColorSpinorField &in) const
+  void DiracDomainWall4D::MdagM(cvector_ref<ColorSpinorField> &out, cvector_ref<const ColorSpinorField> &in) const
   {
     checkFullSpinor(out, in);
-    auto tmp = getFieldTmp(in);
+    auto tmp = getFieldTmp(out);
```

Comment:
What was the reason for switching `in` -> `out` for the `tmp` field? Is it related to the `const`-ness of `in`?

Reply:
exactly, otherwise it creates a constant tmp which is less useful...